acvm-repo/acvm/tests/solver.rs

```diff
@@ -1210,6 +1209,264 @@ where
     fields.into_iter().map(|field| into_repr_vec(field)).collect()
 }
 
+// From https://docs.rs/crate/zkhash/0.2.0/source/ .
+// The original code depends on `ark_ff` v0.4.0, but here we use v0.5.0.
+mod zkhash {
```

Comment:
One quick thought: I'd be more comfortable keeping the external dependency rather than vendoring this code as we're using is precisely to have a reference implementation. We could add an arkworks-0.4 dep and have `run_both_poseidon2_permutations` convert from 0.5 to 0.4 fields internally to get around this.

Reply:
Sorry for being inactive recently due to some personal stuff. Yes, the conversion from ark 0.5 to 0.4 makes a lot of sense because both `zkhash` and `light-poseidon` are used only for tests, and it is definitely better as we no longer need to depend on a pending PR or manually copy the code. Thanks so much for pushing the changes already!